src/core/StelSkyCultureMgr.cpp

```diff
@@ -230,73 +407,142 @@ QString StelSkyCultureMgr::getCurrentSkyCultureHtmlClassification() const
+	{
+		if (singleLicenseForAll)
+			description = q_("This sky culture is provided under Creative Commons License.");
+
+		QStringList details = license.split(" ", SkipEmptyParts);
+
+		const QMap<QString, QString>options = {
+			{ "BY",       q_("You may distribute, remix, adapt, and build upon this sky culture, even commercially, as long "
+			                 "as you credit authors for the original creation.") },
+			{ "BY-SA",    q_("You may remix, adapt, and build upon this sky culture even for commercial purposes, as long as "
+			                 "you credit authors and license the new creations under the identical terms. This license is often "
+			                 "compared to “copyleft” free and open source software licenses.") },
+			{ "BY-ND",    q_("You may reuse this sky culture for any purpose, including commercially; however, adapted work "
+			                 "cannot be shared with others, and credit must be provided by you.") },
+			{ "BY-NC",    q_("You may remix, adapt, and build upon this sky culture non-commercially, and although your new works "
+			                 "must also acknowledge authors and be non-commercial, you don’t have to license your derivative works "
+			                 "on the same terms.") },
+			{ "BY-NC-SA", q_("You may remix, adapt, and build upon this sky culture non-commercially, as long as you credit "
+			                 "authors and license your new creations under the identical terms.") },
+			{ "BY-NC-ND", q_("You may use this sky culture and share them with others as long as you credit authors, but you can’t "
+			                 "change it in any way or use it commercially.") },
+		};
+
+		color = "#33ff33"; // "green" area; free license
+		if (license.contains("ND", Qt::CaseSensitive))
+			color = "#ffff00"; // "yellow" area; nonfree license - weak restrictions
+		if (license.contains("NC", Qt::CaseSensitive))
+			color = "#ff6633"; // "red" area; nonfree license - strong restrictions
+
+		if (!details.at(0).startsWith("CC0", Qt::CaseInsensitive)) // Not public domain!
+			description.append(QString(" %1").arg(options.value(details.at(1), "")));
+		else
+			description = q_("This sky culture is distributed as public domain.");
+	}
+	else if (license.contains("FAL", Qt::CaseSensitive) || license.contains("Free Art License", Qt::CaseSensitive))
+	{
+		color = "#33ff33"; // "green" area; free license
+		description.append(QString(" %1").arg(q_("Free Art License grants the right to freely copy, distribute, and transform.")));
+	}
+
+	return std::make_pair(color, description);
+}
+
+QString StelSkyCultureMgr::getCurrentSkyCultureHtmlLicense() const
+{
+	const auto lines = currentSkyCulture.license.split(QRegularExpression("\\s*\n+\\s*"), SkipEmptyParts);
+	if (lines.isEmpty()) return "";
+
+	switch (lines.size())
+	{
+	case 0:
+	case 1:
+	{
+		const auto line = lines.isEmpty() ? "" : lines[0];
```

Comment:
You have tested lines.isEmpty() above and returned already, right?

Reply:
Indeed, this is a useless check.